stream.go

```diff
@@ -227,7 +238,7 @@ func (s *stream) Write(p []byte) (int, error) {
 			err = errDeadline
 			break
 		}
-		if s.dataForWriting == nil || s.err != nil {
+		if s.dataForWriting == nil || s.canceledWrite || s.closeForShutdownErr != nil {
```

Comment:
I don't think we need the canceledWrite here.

Reply:
Why not? According to the `CancelWrite` doc string, it may be called concurrently with `Write`, and is supposed to make `Write` return immediately. I'm not sure how much sense it make to call those concurrently, but if we want to change it, we should be consistent with the documentation.